### ISSUE TITLE
OpenGL errors while using mpv

### ISSUE THREAD

**Issue by Zaedus** [vaxerski/Hyprland#143]
**Describe the bug**
Some videos that I try to play with mpv throw OpenGL errors. This must have something to do with how mpv interprets certain codecs. Mpv never exists, but it consistently puts out the same errors below.

**Images/videos/anything that would help**
```
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
AV: 00:00:00 / 00:05:25 (0%) A-V: -0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glViewport(0, 0, -349, -196)
[vo/gpu/opengl] GL: GL_INVALID_VALUE in glScissor
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:05:25 (0%) A-V:  0.000
```

Please attach a log (it's in /tmp/hypr/hyprland.log) and, if crashing, a crashdump (coredumpctl, then coredumpctl info <pid>)

*Please do NOT attach coredumps from any packaged version of Hyprland (AUR, etc.). Compile manually, and THEN attach a coredump of that.*

**Comment by vaxerski** [comment]
uhhhh

so you came to the hyprland github, to report a bug in mpv?

**Comment by fufexan** [comment]
He's got a point though. I've tested `mpv` with `-vo=gpu` in both Hyprland and Wayfire. In wayfire all works fine, in Hyprland I get a bunch of
```
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.4s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.5s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.6s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.7s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.7s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.7s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.7s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.6s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.8s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.8s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.
AV: 00:00:00 / 00:03:45 (0%) A-V:  0.000 Cache: 6.8s/1MB
[vo/gpu/opengl] after rendering: OpenGL error INVALID_VALUE.

```

**Comment by Zaedus** [comment]
Oh, that's totally on me! I should have mentioned that I only experienced this with Hyprland and no other Wayland compositor.

**Comment by vaxerski** [comment]
ok, I mean mpv worked for me, just had some issues with the bar. Will look into.

In the meantime, just use the superior vlc.

**Comment by ThatOneCalculator** [comment]
Might be related to #163 ?

**Comment by vaxerski** [comment]
fixed in d73e05cc681cb7227d69d67c32e735b2705743a6
